modules/service/permissioned-data-service.js

```diff
@@ -0,0 +1,86 @@
+const BN = require('bn.js');
+
+const Utilities = require('../Utilities');
+const Models = require('../../models');
+
+class PermissionedDataService {
```

Comment:
Can you make this to be controller, since it is called from rest-api. We agreed that controllers should be the ones that handle requests.